python/pyspark/sql/tests.py

```diff
@@ -1220,6 +1220,13 @@ def test_rand_functions(self):
         rndn2 = df.select('key', functions.randn(0)).collect()
         self.assertEqual(sorted(rndn1), sorted(rndn2))
 
+    def test_string_functions(self):
+        from pyspark.sql.functions import col, lit
+        df = self.spark.createDataFrame([['nick']], schema=['name'])
+        self.assertRaises(TypeError, lambda: df.select(col('name').substr(0, lit(1))))
```

Comment:
I was considering doing that at first, but it felt like just duplicating logic. Looking through the other uses of `assertRaisesRegexp()`, it looks like most of the time we just search for a keyword, but there are also some instances where a large part of the exception message is checked. I can do that here as well.